src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -357,6 +357,18 @@ private function validateAndDenormalize(string $currentClass, string $attribute,
         throw new NotNormalizableValueException(sprintf('The type of the "%s" attribute for class "%s" must be one of "%s" ("%s" given).', $attribute, $currentClass, implode('", "', array_keys($expectedTypes)), gettype($data)));
     }
 
+    /**
+     * @return Type[]|null
+     */
+    protected function getTypes(string $currentClass, string $attribute)
```

Comment:
Can you make it `@internal`? The proliferation of public methods in this class make it very hard to maintain (we should split it in several classes/traits instead).